btf/btf.go

```diff
@@ -694,3 +761,55 @@ func (iter *TypesIterator) Next() bool {
 	iter.index++
 	return true
 }
+
+type mergedSpec []*Spec
+
+func (s mergedSpec) TypeID(typ Type) (TypeID, error) {
+	for _, sp := range s {
+		id, err := sp.TypeID(typ)
+		if err != nil {
+			if errors.Is(err, ErrNotFound) {
+				continue
+			}
+			return 0, err
+		}
+		return id, nil
+	}
+	return 0, ErrNotFound
+}
+
+func (s mergedSpec) NamedTypesIterate(name essentialName) *namedTypesIterator {
+	iter := &namedTypesIterator{specIndex: 0, typeIndex: 0}
+	for _, sp := range s {
+		namedTypes := sp.namedTypes[name]
+		if len(namedTypes) > 0 {
+			iter.types = append(iter.types, namedTypes)
```

Comment:
This is not going to work any more with lazy Spec.Copy going in. Iterating all the named types is "expensive" since it now causes copying.